### DISCUSSION TITLE
Cisco ASA Logs not assigning time correctly

### DISCUSSION THREAD

**Discussion by Git-Me-Some-Hub** [Security-Onion-Solutions/securityonion#5390] [Unsupported Versions] [answered]
ISO - Version: 2.3.51 - I am using the built-in ES Modules

I am having an issue that the ASA syslogs that are coming in do not assign the proper time to the data. From the Message field I am not seeing it included, so I can only assume that SO-ElasticSearch is assigning it based on ingestion time. I see in Kibana that the times are supposed to change based on your browser time, and I see the option to change it in Stack Management but it seem correct. So my only assumption is that it is SO not adding a timezone some something to the field, so Kibana does not know how to interpret it? I just don't know. 

Here is a dashboard that shows the different data set that are coming in overtime. It currently has a bit of ASA files assigned for the 10am timeframe, while it is actually after 4pm. That is a 6 hour difference. As you can also see the other test logs I am ingesting are being assigned properly, just not the ASA. The ASA is calling to the same NTP server as SO, and was confirmed to match on the time. 
![image](https://user-images.githubusercontent.com/17930434/131922443-7e882abf-2bb6-4c25-8d9f-a987fc1b7db1.png)

This is a specific record that I was seeing yesterday. 
Timestamp:
![image](https://user-images.githubusercontent.com/17930434/131922706-ff7a0fc2-47fa-425f-873f-3e7287e7a851.png)
Ingestion Time:
![image](https://user-images.githubusercontent.com/17930434/131922777-6dd59c3d-5293-41a7-840d-3668d00f49be.png)
Time zone: 
![image](https://user-images.githubusercontent.com/17930434/131922826-a1773d13-5015-4877-b86f-d8f36db7786f.png)

Not sure if this is raw log or not:
![image](https://user-images.githubusercontent.com/17930434/131922972-bfc634aa-04e3-466e-a148-e7fb362388e1.png)

**Comment by dougburks** [comment]
If you are using version 2.3.51, then the first thing I would recommend is updating to 2.3.70 to make sure that you're using the latest Elastic modules.

**Reply by Git-Me-Some-Hub** [reply]
Thanks Doug, I did go ahead and update this last week after posting this... like you said to make sure. Confirmed it was at the latest today; however, still the same issue. 

Current time is just after 3pm: 
![image](https://user-images.githubusercontent.com/17930434/132411155-92ea7259-cc20-4469-ac1e-84303c2e537f.png)

**Comment by dougburks** [comment, marked as answer]
You mention a 6 hour difference. What time zone are you in?

**Reply by Git-Me-Some-Hub** [reply]
Mountain Time zone.

**Reply by dougburks** [reply]
OK, so would the difference between Mountain Time zone and UTC explain the 6 hour difference?
https://docs.securityonion.net/en/2.3/timezones.html

**Reply by Git-Me-Some-Hub** [reply]
Well I am not sure I follow. I do know that mountain is 6 hours difference, but it actually should be 6 greater(later) than mountain, right? So if it is midnight 00:00 in mountain it should be 06:00 UTC? However, it is doing the exact opposite. 

That is also leaving out the fact that it says Kibana should be interpreting the browsers time zone and changing it accordingly. Which does seem to be happening with all logs except the ASA. 

Any idea why that might be?

**Reply by dougburks** [reply]
Is it possible that perhaps the ASA is configured for local time and sends a timestamp without time zone information and then Elasticsearch assumes that timestamp is UTC?

**Reply by Git-Me-Some-Hub** [reply]
Yeah, you did hit the nail on the head there. I did a TCPdump and saw that it was coming in as mountain time, but no time zone included. 

My Network Admin changed the log output to RFC 5424, and that includes the Time zone.

Thanks for working this out with me to find the right answers.
